backend/layers/persistence/persistence.py

```diff
@@ -0,0 +1,153 @@
+from datetime import datetime
+from typing import List
+from backend.corpora.common.entities.dataset import Dataset
+from backend.layers.common.entities import CollectionMetadata, CollectionVersion, DatasetArtifact, DatasetMetadata, DatasetStatus, DatasetVersion
+
+
+class DatabaseProviderInterface:
+        
+    def initialize_canonical_collection(self, owner: str, collection_metadata: CollectionMetadata) -> CollectionVersion:
+        """
+        Initializes a canonical collection, generating a canonical collection_id and a new version_id.
+        Returns the newly created CollectionVersion
+        """
+        pass
+
+
+    def get_collection_version(self, version_id: str) -> CollectionVersion:
+        """
+        Retrieves a specific collection version by id
+        """
+        pass
+
+    def get_all_versions_for_collection(self, collection_id: str) -> List[CollectionVersion]:
+        """
+        Retrieves all versions for a specific collections, without filtering
+        """
+        pass
+
+    def get_all_collections(self) -> Iterable[CollectionVersion]:
```

Comment:
would rename:
```suggestion
    def get_all_collections_latest_version(self) -> Iterable[CollectionVersion]:
```
I see `TODO: add a filtering object`.  This will allow client to get just published collections, right?

Reply:
This method needs to be more generic than just `latest_version`. For instance, we also need to retrieve all the private collections (aka, all the collection versions that haven't been previously published) owned by someone, or by all users (in case of a super curator), hence the necessity of a filtering object. Your method still makes sense and can be added as a secondary method that relies on this one.

Reply:
But it should only ever return a single collection_version for any given collection_id, right? If so, the method doesn't capture these semantics.

Could the return type be `Dict[CollectionId, CollectionVersion]`?  (we'd have to define `CollectionId`)